apps/web/src/components/agent/chats.tsx

```diff
@@ -56,16 +56,17 @@ export const getPublicChatLink = (agentId: string, workspace: Workspace) => {
 function Page() {
   const [params] = useSearchParams();
 
-  const { agentId, workspace, threadId } = useMemo(() => {
+  const { agentId, workspace, threadId, inlineMcps } = useMemo(() => {
     const workspace = params.get("workspace") as Workspace | null;
     const agentId = params.get("agentId");
+    const inlineMcps = params.getAll("inlineMcp");
```

Comment:
two different ways:

1. receive as toolsetsBase64=encodeURIComponent(btoa(JSON.stringify(toolsets))
2. decode here

1. receive as toolsets=https://url,SSE
2. fallback to HTTP